qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -246,7 +252,18 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
                 )
 
             all_circuit_results.append(circuit_results)
-            oracle_evaluation = amplification_problem.is_good_state(top_measurement)
+
+            if len(iterations) == 1 and amplification_problem.is_good_state is None:
+                oracle_evaluation = None  # cannot check for good state without is_good_state arg
```

Comment:
Should we add a didactic `break` statement here to make clear we're done? Theoretically it's captured by the `if oracle_evaluation` below but that one looks like it's made for bools.
```suggestion
                oracle_evaluation = None  # cannot check for good state without is_good_state arg
                break
```